From 8b8ef7ea0a6ebb3ecbd7e15543cd0fc662f8173d Mon Sep 17 00:00:00 2001
From: Vladimir Oltean <olteanv@gmail.com>
Date: Sun, 6 Jan 2019 19:38:31 +0200
Subject: [PATCH 30/31] extract_utils: be compatible with system-as-root
 layouts

* The use case is that if you have the following layout:
  $TOP --- system.img
       |
       +-- vendor.img
  you should be able (from $TOP) to:
    mkdir system; mount -o ro,loop system.img system
    mkdir vendor; mount -o ro,loop vendor.img vendor
  and then (from device tree)
    ./extract-files.sh $TOP

  But this doesn't work if system.img is SAR and contains another
  "system" dir inside. This patch makes sure it searches for a "system"
  dir in the provided path as well, if it couldn't find the blob
  anywhere else.

Change-Id: Ib49cd5b587b3a57478a66ff69cf840270c2b1403
Signed-off-by: Vladimir Oltean <olteanv@gmail.com>
---
 build/tools/extract_utils.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/tools/extract_utils.sh b/build/tools/extract_utils.sh
index 30fa8a6a..6b79f244 100644
--- a/build/tools/extract_utils.sh
+++ b/build/tools/extract_utils.sh
@@ -767,6 +767,7 @@ function get_file() {
         # try to copy
         cp -r "$SRC/$1"           "$2" 2>/dev/null && return 0
         cp -r "$SRC/${1#/system}" "$2" 2>/dev/null && return 0
+        cp -r "$SRC/system/$1"    "$2" 2>/dev/null && return 0
 
         return 1
     fi
-- 
2.17.1

